src/binding.rs

```diff
@@ -184,15 +197,19 @@ impl<I: Into<Binding>> InputModKeys for I {
     ///
     /// # Panics
     ///
-    /// Panics when called on [`Binding::GamepadButton`], [`Binding::GamepadAxis`] or [`Binding::None`].
+    /// Panics when called on [`Binding::GamepadButton`], [`Binding::GamepadAxis`],
+    /// [`Binding::AnyKey`] or [`Binding::None`].
     fn with_mod_keys(self, mod_keys: ModKeys) -> Binding {
         match self.into() {
             Binding::Keyboard { key, .. } => Binding::Keyboard { key, mod_keys },
             Binding::MouseButton { button, .. } => Binding::MouseButton { button, mod_keys },
             Binding::MouseMotion { .. } => Binding::MouseMotion { mod_keys },
             Binding::MouseWheel { .. } => Binding::MouseWheel { mod_keys },
-            Binding::GamepadButton { .. } | Binding::GamepadAxis { .. } | Binding::None => {
-                panic!("keyboard modifiers can be applied only to mouse and keyboard")
+            Binding::GamepadButton { .. }
+            | Binding::GamepadAxis { .. }
+            | Binding::None
+            | Binding::AnyKey => {
+                panic!("keyboard modifiers can be applied only to mouse and keyboard bindings")
```

Comment:
Non-blocking: this should not panic. It is fully recoverable: just log a warning and then retrun `Self`.